src/delegation/agent.rs

```diff
@@ -73,7 +72,7 @@ where
         let nonce = Nonce::generate_12(&mut salt);
 
         if let Some(ref sub) = subject {
-            if sub == self.did {
+            if sub == &self.did {
```

Comment:
Minor question about this, because I run into it a lot. There's two options:

```rust
if sub == &self.did
// vs
if *sub == self.did
```

I assume that when you compare with `&`s, it has to deref these (which is extra indirection). The compiler can probably get rid of this, but do we know?

Reply:
Hm. Good question.
In the error message the compiler suggests `&`, but `*` totally works here, too.